modules/fedimint-wallet/src/lib.rs

```diff
@@ -193,9 +193,9 @@ impl Serialize for UnsignedTransaction {
 }
 
 struct StatelessWallet<'a> {
-    descriptor: &'a Descriptor<CompressedPublicKey>,
-    secret_key: &'a secp256k1::SecretKey,
-    secp: &'a secp256k1::Secp256k1<secp256k1::All>,
+    pub descriptor: &'a Descriptor<CompressedPublicKey>,
+    pub secret_key: &'a secp256k1::SecretKey,
+    pub secp: &'a secp256k1::Secp256k1<secp256k1::All>,
```

Comment:
Why do these need to be pub? (not that we'd have great pub hygiene elsewhere, but would be nice to become more restrictive over time to reduce public APIs we are committed to)

Reply:
To be constructed in the `mod tests`.  Is there a better way around this?

Reply:
Can't submodules always access private fields of structs in super? Maybe I'm completely off here.

Reply:
They should, AFAIK.

There's also `pub(crate)` and for integration tests we can always make some "hidden constructor" `#[doc(hidden)]` to cheat, yet not expose publicly stuff we don't want as a stable API.

Reply:
Ah you are right, not sure why I got a compile error earlier.